crates/driver/src/logic/competition/solution/mod.rs

```diff
@@ -1,131 +1,117 @@
 use {
     crate::{
+        blockchain,
         boundary,
-        logic::{competition, eth},
+        logic::{
+            competition::{self, order},
+            eth,
+        },
         simulator,
         solver::{self, Solver},
-        util,
         Ethereum,
         Simulator,
     },
+    futures::future::try_join_all,
+    itertools::Itertools,
     num::ToPrimitive,
     primitive_types::U256,
     std::collections::HashMap,
 };
 
-mod approvals;
-pub mod settlement;
+pub mod interaction;
+mod settlement;
+pub mod trade;
 
-pub use {approvals::Approvals, settlement::Settlement};
+pub use {interaction::Interaction, settlement::Settlement, trade::Trade};
 
 /// A solution represents a set of orders which the solver has found an optimal
 /// way to settle. A [`Solution`] is generated by a solver as a response to a
 /// [`super::auction::Auction`].
 #[derive(Debug)]
 pub struct Solution {
-    // TODO After #831 we hope to make this a vec and not a hashmap
-    pub orders: HashMap<usize, Order>,
-    pub jit_orders: Vec<JitOrder>,
-    // TODO This will be changed in #831:
-    // 1. It should not be a HashMap, it should be a list.
-    // 2. It should be part of the orders field, as an enum
-    // But I'll make it work like this for now
-    pub amms: HashMap<eth::Address, Vec<Amm>>,
-    pub prices: HashMap<eth::Token, eth::Ether>,
-    pub approvals: Approvals,
-    pub interactions: Vec<SafeInteraction>,
+    /// Trades settled by this solution.
+    pub trades: Vec<Trade>,
+    /// Token prices for this solution.
+    pub prices: HashMap<eth::TokenAddress, competition::Price>,
+    /// List of interactions with smart contracts needed to execute this
+    /// solution on the blockchain.
+    pub interactions: Vec<Interaction>,
```

Comment:
In theory, interactions are split into pre-, intra-, and post- interactions. While this doesn't matter for the `driver` <> `solver/searcher` DTO, I think this matters for the Solution domain object (since we already make use of pre-interactions for example for EthFlow contracts).

Reply:
Yes, it is needed when building the settlement calldata. The `searcher/solver` doesn't care about this because their interactions always go in the middle, but when including things like EthFlow pre-interactions when building settlement data for the driver to submit on-chain, it would need to care about this.